lib/curl_sha512_256.c

```diff
@@ -0,0 +1,596 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) Evgeny Grin (Karlson2k), <k2k@narod.ru>.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * SPDX-License-Identifier: curl
+ *
+ ***************************************************************************/
+
+#include "curl_setup.h"
+
+#if !defined(CURL_DISABLE_DIGEST_AUTH) && !defined(CURL_DISABLE_SHA512_256)
+
+#include "curl_sha512_256.h"
+#include "warnless.h"
+
+/* ** This implementation of SHA-512/256 hash calculation was originally ** *
+ * ** written by Evgeny Grin (Karlson2k) for GNU libmicrohttpd.          ** *
+ * ** The author ported the code to libcurl. The ported code is provided ** *
+ * ** under curl license.                                                ** *
+ * ** This is a minimal version with minimal optimisations. Performance  ** *
+ * ** can be significantly improved. Big-endian store and load macros    ** *
+ * ** are obvious targets for optimisation.                              ** *
+ * ** The code is designed in a way so any wide enough type (64 bits or  ** *
+ * ** more) can be used for calculations. Potentially it can be larger   ** *
+ * ** then 64 bits.                                                      ** */
```

Comment:
This comment is not adding much for the reader and does not need to be provided.

```suggestion
```

Reply:
The same (or quite similar) code was already published under LGPL-2.1+ license.
As I wrote the code alone, I can apply another license.

Similar comments exist in [md5.c](https://github.com/curl/curl/blob/5a4b2f93392c0fb1171dd17b67b05f39c6d45abc/lib/md5.c#L254) and [md4.c](https://github.com/curl/curl/blob/5a4b2f93392c0fb1171dd17b67b05f39c6d45abc/lib/md4.c#L228) code algorithms. I decided to add comment here to solve any possible questions about getting the code from incompatible license (`curl` license is compatible with `LGPL` but `LGPL` code cannot be re-used directly under `curl` license, unless the original author changed the license).